modules/http4s-swagger/src-jvm/smithy4s/http4s/swagger/Docs.scala

```diff
@@ -18,46 +18,52 @@ package smithy4s
 package http4s
 package swagger
 
+import cats.data.NonEmptyList
 import cats.effect.Sync
-import org.http4s.HttpRoutes
 import org.http4s._
 import org.http4s.dsl.Http4sDsl
 import org.http4s.headers.Location
+import org.http4s.HttpRoutes
 import org.webjars.WebJarAssetLocator
 
 private[smithy4s] abstract class Docs[F[_]](
-    hasId: HasId,
+    ids: NonEmptyList[HasId],
     path: String,
     swaggerUiPath: String
 )(implicit F: Sync[F])
     extends Http4sDsl[F]
     with Compat.DocsClass[F] {
 
-  val jsonSpec = hasId.id.namespace + '.' + hasId.id.name + ".json"
+  private def toSwaggerUrl(id: HasId): (String, SwaggerInit.SwaggerUrl) = {
+    val jsonSpec = id.id.namespace + '.' + id.id.name + ".json"
+    jsonSpec -> SwaggerInit.SwaggerUrl(s"/$jsonSpec", id.id.name)
+  }
+  private val validSpecs = ids.map(toSwaggerUrl).map(_._1).toList
+  private val specsUrls = ids.map(toSwaggerUrl).map(_._2)
 
   val actualPath: Path = Uri.Path.unsafeFromString("/" + path)
 
   object DocPath {
     def unapply(p: Path): Boolean = {
       p match {
-        case `actualPath`                                               => true
-        case `actualPath` / ""                                          => true
-        case `actualPath` / file if file.equalsIgnoreCase("index.html") => true
-        case _                                                          => false
+        case `actualPath`      => true
+        case `actualPath` / "" => true
+        case _                 => false
       }
     }
   }
   def routes: HttpRoutes[F] = HttpRoutes.of[F] {
     case r @ GET -> DocPath() if r.uri.query.isEmpty =>
-      PermanentRedirect(
-        Location(Uri.unsafeFromString(s"/$path/index.html?url=/$jsonSpec"))
-      )
+      Found(Location(Uri.unsafeFromString(s"/$path/index.html")))
```

Comment:
I'm curious (and also somewhat illiterate when it comes to 3xx statuses) : can you tl;dr the "why" of changing the status ?

Reply:
I changed it because 308 is cached very aggressively (forever) and thus you can't change a redirect afterwards. But I can revert that, I don't have an opinion on the matter. On quick look at MDN suggest that it's mostly for SEO purposes.

In my case, it was annoying because I had run a smithy project locally before, accessed the docs (for say PizzaAdmin) and now my browser was always redirecting me to this old url with pizza spec in it

Reply:
Makes total sense, thanks !